pkg/types/azure/machinepool.go

```diff
@@ -50,6 +52,12 @@ type MachinePool struct {
 	// +optional
 	VMNetworkingType string `json:"vmNetworkingType,omitempty"`
 
+	// BootDiagnostics has the value for the storage account URI where the
+	// machine log information is sent to. Defaults to managed if no value
+	// is mentioned.
```

Comment:
I think this is really defaults to managed for control plane, disabled for compute nodes